### PR TITLE
Fix CI by imitating Mojolicious .travis.yml, fix memory cycles

### PR DESCRIPTION
**Pull request by mohawk2**

### CODE REVIEW

**Comment by mohawk2** [comment]
Sorry, I meant to add: Travis isn't running your CI currently. I believe you just need to go to https://travis-ci.org/github/jhthorsen to re-enable it.

**Comment by mohawk2** [comment]
Also, 5.10 (only) is making a bunch of uninitialised value warnings: https://travis-ci.org/github/mohawk2/mojo-redis/jobs/685067228

**Comment by jhthorsen** [comment]
I've replaced travis with github workflow now. Also, there's so many changes and commits here that it's hard to follow. Please clean up and make a new PR.
